src/circuit/operator/generator.rs

```diff
@@ -8,48 +8,55 @@ use std::borrow::Cow;
 /// seed and a transformer function.  On `clock_start`, the internal operator
 /// state is reset to the seed value.  On each `eval`, the operator outputs
 /// its current state before updating the state using the transformer function.
-pub struct Generator<T, F> {
-    seed: T,
+pub struct Generator<T, S, F> {
+    current: T,
     val: T,
+    state: S,
     update: F,
 }
 
-impl<T, F> Generator<T, F>
+impl<T, S, F> Generator<T, S, F>
 where
     T: Clone,
 {
-    /// Creates a generator with initial `seed` and state transformer function.
-    pub fn new(seed: T, update: F) -> Self {
+    /// Creates a generator with initial value `current` and state transformer function.
+    pub fn new(current: T, state: S, update: F) -> Self {
         Self {
-            seed: seed.clone(),
-            val: seed,
+            current: current.clone(),
```

Comment:
The naming is of fields is confusing. It seems that `current` should be called `initial`,  and `val` should be called `output`.
Also, do we need to store the `output` inside the operator?  We already have `state` for anything that should survive till the next clock tick.

Reply:
An alternative way to generalize this operator is to remove all state except closure and have the closure capture any state variables upon operator creation.  I should have thought of that earlier.